### PR TITLE
Swap out Decapodes dep for DiagEqs

### LINKED ISSUE #32: SyntacticModels `main` broken when building against Decapodes `main`
Latest SyntacticModels doesn't compile with latest Decapodes

## Steps to reproduce
```
julia> ] add https://github.com/AlgebraicJulia/SyntacticModels.jl/
julia> ] add https://github.com/AlgebraicJulia/Decapodes.jl/

```
## Output
```
PkgPrecompileError: The following 1 direct dependency failed to precompile:

SyntacticModels [22bb929c-8bcf-4852-b455-eb3e1675e09c]

Failed to precompile SyntacticModels [22bb929c-8bcf-4852-b455-eb3e1675e09c] to "/home/five/.julia/compiled/v1.10/SyntacticModels/jl_u65HBr".
ERROR: LoadError: UndefVarError: `Equation` not defined
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base ./Base.jl:31
 [2] top-level scope
   @ ~/.julia/packages/SyntacticModels/hL8gg/src/decapodes.jl:41
 [3] include(mod::Module, _path::String)
   @ Base ./Base.jl:495
 [4] include(x::String)
   @ SyntacticModels ~/.julia/packages/SyntacticModels/hL8gg/src/SyntacticModels.jl:1
 [5] top-level scope
   @ ~/.julia/packages/SyntacticModels/hL8gg/src/SyntacticModels.jl:5
 [6] include
   @ Base ./Base.jl:495 [inlined]
 [7] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2216
 [8] top-level scope
   @ stdin:3
in expression starting at /home/five/.julia/packages/SyntacticModels/hL8gg/src/decapodes.jl:1
in expression starting at /home/five/.julia/packages/SyntacticModels/hL8gg/src/SyntacticModels.jl:1
in expression starting at stdin:
```

### PR DESCRIPTION
**Pull request by lukem12345**
Close #32

### CODE REVIEW

**Comment by lukem12345** [comment]
@jpfairbanks If tests pass this will be good to merge upon your review.

**Comment by lukem12345** [comment]
@jpfairbanks I will get back to working on this at 4:15, but if the fix is obvious to you, you can push here in the meantime.

**Comment by lukem12345** [comment]
@jpfairbanks 
Tests pass locally. These are the contents of `harmonic_oscillator.json`. Please advise on if this is the JSON we want:
```JSON
{
    "annotations": [
        {
            "note": {
                "str": "The X variable.",
                "_type": "Name"
            },
            "type": "Form0",
            "entity": "X",
            "_type": "Annotation"
        }
    ],
    "header": {
        "description": "A Simple Harmonic Oscillator as a Diagrammatic Equation",
        "name": "harmonic_oscillator",
        "_type": "Header",
        "model_version": "v1.0",
        "schema": "modelreps.io/DecaExpr",
        "schema_name": "DecaExpr"
    },
    "_type": "ASKEMDecaExpr",
    "model": {
        "context": [
            {
                "var": "X",
                "dim": "Form0",
                "space": "Point"
            },
            {
                "var": "V",
                "dim": "Form0",
                "space": "Point"
            },
            {
                "var": "k",
                "dim": "Constant",
                "space": "Point"
            }
        ],
        "_type": "DecaExpr",
        "equations": [
            {
                "rhs": {
                    "name": "V",
                    "_type": "Var"
                },
                "lhs": {
                    "var": {
                        "name": "X",
                        "_type": "Var"
                    },
                    "_type": "Tan"
                },
                "_type": "Eq"
            },
            {
                "rhs": {
                    "args": [
                        {
                            "name": "-1",
                            "_type": "Lit"
                        },
                        {
                            "name": "k",
                            "_type": "Var"
                        },
                        {
                            "name": "X",
                            "_type": "Var"
                        }
                    ],
                    "_type": "Mult"
                },
                "lhs": {
                    "var": {
                        "name": "V",
                        "_type": "Var"
                    },
                    "_type": "Tan"
                },
                "_type": "Eq"
            }
        ]
    }
}
```

File: src/SyntacticModelsBase.jl [hunk 1]
```diff
@@ -23,6 +24,7 @@ _dict(x::Symbol) = x
 _dict(x::String) = x
 _dict(x::Number) = x
 _dict(x::AbstractVector) = map(_dict, x)
+_dict(x::DiagrammaticEquations.decapodes.Judgement) = x
```
Comment (lukem12345) [inline src/SyntacticModelsBase.jl:27]: @jpfairbanks This line feels suspect to me, but is what got the JSON to generate and tests to pass.

**Comment by jpfairbanks** [comment]
LGTM

**Review by jpfairbanks — APPROVED** [review]

**Comment by lukem12345** [comment]
Swapped in DiagrammaticEquations in the docs. Will merge whenever they pass.

**Pull request merged**